web/react/components/user_settings/user_settings_general.jsx

```diff
@@ -106,15 +109,33 @@ export default class UserSettingsGeneralTab extends React.Component {
             return;
         }
 
+        if (email !== confirmEmail) {
+            this.setState({emailError: 'The new emails you entered do not match'});
+            return;
+        }
+
         user.email = email;
 
-        this.submitUser(user);
+        if (this.state.emailEnabled && this.state.emailVerificationEnabled) {
+            this.submitUser(user, {emailChangeInProgress: true});
+        } else {
+            this.submitUser(user, {emailChangeInProgress: false});
+        }
     }
-    submitUser(user) {
+    submitUser(user, newState) {
```

Comment:
There was a point when I was trying to do the original spec that this made more sense :P but you're right it's not a simple way to do it now. I should just be able to check for whether email and verification is enabled within the submitUser function and do it there, I'll change it shortly, thanks!